rudder-core/src/main/scala/com/normation/rudder/domain/nodes/Node.scala

```diff
@@ -135,22 +180,41 @@ object CompareProperties {
    *   the property
    * - if the value is the emtpy string, remove
    *   the property
+   *
+   * Each time, we have to check the provider of the update to see if it's compatible.
+   * Node that in read-write mode, the provider is the last who wrote the property.
    */
-  def updateProperties(props: Seq[NodeProperty], updates: Option[Seq[NodeProperty]]) = {
+  def updateProperties(props: Seq[NodeProperty], updates: Option[Seq[NodeProperty]], provider: NodePropertyProvider): Box[Seq[NodeProperty]] = {
```

Comment:
it's used in the comparison to check in the read-only case: the same provider can modify, whatever the access mode.